theories/Logic/FunctionalExtensionality.v

```
@@ -68,6 +68,25 @@ Tactic Notation "extensionality" ident(x) :=
      apply forall_extensionality) ; intro x
   end.
 
+(** Iteratively apply [functional_extensionality] on an hypothesis
+    until finding an equality statement *)
+
+Tactic Notation "extensionality" "in" hyp(H) :=
+  let faildep _ := fail "Not a non-dependent extensional equality" in
+  let rec iter_fun_ext :=
+    match type of H with
```

Do you mean `lazymatch`, here?  Else your failure from `faildep` at level 0 will be swallowed by match backtracking, as will any recursive failures.